climada/engine/impact.py

```diff
@@ -1409,123 +1417,177 @@ def concat(cls, imp_list,concat_type='time_window'):
                New impact object which is a concatenation of all impacts
         """
 
+        # checks
+        def check_haz_types(imp_list):
+            haz_types = [imp.tag['haz'].haz_type for imp in imp_list if \
+                         imp.tag['haz'].haz_type != '']
+            if len(haz_types) < len(imp_list):
+                raise ValueError(
+                    "Not all impacts have a hazard type."
+                    "Please specify the same hazard type for each impact."
+                    "The impacts are incompatible and cannot be concatenated.")
+            if len(set(haz_types)) > 1:
+                raise ValueError(f"Impacts are based on hazards of different types: {haz_types}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            return True
+
+        def check_units(imp_list):
+            units = [imp.unit for imp in imp_list if imp.unit != '']
+            if len(units) == 0:
+                LOGGER.warning("No units specified. Assume unitless impact measure.")
+            elif len(units) < len(imp_list):
+                raise ValueError("Some impacts have a unit, some not. Same units needed."
+                                 "The impacts are incompatible and cannot be concatenated.")
+            if len(set(units)) > 1:
+                raise ValueError(f"The given impacts have different units: {units}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            unit = set(units[0])
+            return unit
+
+        def check_crs(imp_list):
+            crs = [imp.crs for imp in imp_list if imp.crs != '']
+            if len(crs) < len(imp_list):
+                raise ValueError(
+                    "Not all impacts have a CRS. Specify the same CRS for each impact."
+                    "The impacts are incompatible and cannot be concatenated.")
+            if len(set(crs)) > 1:
+                raise ValueError(f"The given impacts have different crs: {crs}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            return crs[0]
+
+        def check_attributes(imp_list):
+            attributes = set.union(*[set(vars(imp).keys()) for imp in imp_list])
+            for attr_name in attributes:
+                if not all(hasattr(imp, attr_name) for imp in imp_list):
+                    raise ValueError(f"Attribute {attr_name} is not shared by all impacts. "
+                                    "The impacts are incompatible and cannot be concatenated.")
+            return True
+
+        def check_frequency(imp_list):
+            freqs = {imp.frequency[0] for imp in imp_list}
+            if len(freqs) > 1:
+                raise ValueError(f"Impacts do not have the same frequency: {freqs}."
+                                 "The impacts are incompatible and cannot be concatenated.")
```

Comment:
Impacts do not have a frequency... do you mean the hazard events associated with the given impact? Or what is meant here? 

Also, I am a still bit confused why you need the same frequencies to concatenate. You just need the full impact matrices if the frequencies of the events are different right?

Reply:
You are probably right regarding your point with the frequencies. I remove the frequency check.